### ISSUE TITLE
Invalid gemspec in [.......]: stack level too deep

### ISSUE THREAD

**Issue by alimp5** [hashicorp/vagrant#9788]
Hi Vagrant Team,

### Vagrant version
Vagrant 2.0.2

### Ruby version
ruby 2.5.1p57 (2018-03-29 revision 63029) [x86_64-linux-gnu]

### Host operating system
Linux kali 4.15.0-kali2-amd64 #1 SMP Debian 4.15.11-1kali1 (2018-03-21) x86_64 GNU/Linux
I upgraded Kali to the last changes via `dist-upgrade` and `upgrade` with "apt-get" command.

### Vagrantfile
as default. without modification.

### Virtualbox
VirtualBox Graphical User Interface Version 5.2.8_Debian r120774
© 2004-2018 Oracle Corporation (Qt5.9.2)

### behavior
```
root@kali:~# vagrant box add ubuntu/xenial64
Invalid gemspec in [/usr/share/rubygems-integration/all/specifications/rbnacl-libsodium.gemspec]: stack level too deep
Invalid gemspec in [/usr/share/rubygems-integration/all/specifications/rbnacl-libsodium.gemspec]: stack level too deep
==> box: Loading metadata for box 'ubuntu/xenial64'
    box: URL: https://vagrantcloud.com/ubuntu/xenial64
==> box: Adding box 'ubuntu/xenial64' (v20180504.0.0) for provider: virtualbox
    box: Downloading: https://vagrantcloud.com/ubuntu/boxes/xenial64/versions/20180504.0.0/providers/virtualbox.box
==> box: Successfully added box 'ubuntu/xenial64' (v20180504.0.0) for 'virtualbox'!
root@kali:~# vagrant box list
Invalid gemspec in [/usr/share/rubygems-integration/all/specifications/rbnacl-libsodium.gemspec]: stack level too deep
Invalid gemspec in [/usr/share/rubygems-integration/all/specifications/rbnacl-libsodium.gemspec]: stack level too deep
root@kali:~# 
```
Why the list is empty !??
Why I got this error or warning message for every vagrant command !?
How to fix it !?

Thanks in advance

**Comment by chrisroberts** [comment]
Hi there,

This looks like a problem with your Vagrant installation. The specification warnings from the rbnacl-libsodium gemspec are not associated with Vagrant. How did you install Vagrant? Please also provide a gist of the output from: `vagrant box list --debug`.

Thanks!

**Comment by Ajay2007** [comment]
Hey please will somebody give the answer for this question , I am having same problem too.

**Comment by alimp5** [comment]
Hi @chrisroberts

any update on topic !?

Thanks

**Comment by egonyuri** [comment]
Heys Guys I found a solution, 

Just update your vagrant that works. Download direct from the vagrant website and don't use the version of ubuntu repository (apt-get install vagrant). The ubuntu version is outdated.

**Comment by chrisroberts** [comment]
Hi there,

@egonyuri is correct. Using the installer from the [downloads page](https://www.vagrantup.com/downloads.html) will resolve this issue. I'm not sure how you have Vagrant installed, but it does not appear to be from the official installers.

Cheers!

**Comment by gugacavalieri** [comment]
thanks @egonyuri 

That solved it :+1: :smile:

**Comment by stadja** [comment]
3 months later it still solved  it @egonyuri : you rock.

**Comment by briancain** [comment]
Hey @VickySteeves - sorry the Ubuntu packages are broken! Unfortunately we at HashiCorp have no control over what version of Vagrant Ubuntu ships with, and therefore is not officially supported by us.

**Comment by rzr** [comment]
It looks like it's affecting vagrant shipped in debian

**Comment by ginsburgnm** [comment]
Currently it looks like Hashicorp packages are also broken. Neither installation works, the Ubuntu packages actually seem less broken at the moment. because it's at least trying with the stack level too deep error. The hashicorp package just says it finds no vagrant executable.

**Comment by chrisroberts** [comment]
@ginsburgnm Hi there. The current release of Vagrant functions as expected on Ubuntu. I just spun up a fresh system to validate:

```
vagrant@vagrant:~$ wget -q https://releases.hashicorp.com/vagrant/2.2.0/vagrant_2.2.0_x86_64.deb
vagrant@vagrant:~$ cat /etc/lsb-release 
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=18.04
DISTRIB_CODENAME=bionic
DISTRIB_DESCRIPTION="Ubuntu 18.04 LTS"
vagrant@vagrant:~$ sudo dpkg -i ./vagrant_2.2.0_x86_64.deb 
Selecting previously unselected package vagrant.
(Reading database ... 41996 files and directories currently installed.)
Preparing to unpack ./vagrant_2.2.0_x86_64.deb ...
Unpacking vagrant (1:2.2.0) ...
Setting up vagrant (1:2.2.0) ...
vagrant@vagrant:~$ vagrant
WARNING: This command has been deprecated in favor of `vagrant cloud auth login`
Usage: vagrant [options] <command> [<args>]

    -v, --version                    Print the version and exit.
    -h, --help                       Print this help.

Common commands:
     box             manages boxes: installation, removal, etc.
     cloud           manages everything related to Vagrant Cloud
     destroy         stops and deletes all traces of the vagrant machine
     global-status   outputs status Vagrant environments for this user
     halt            stops the vagrant machine
     help            shows the help for a subcommand
     init            initializes a new Vagrant environment by creating a Vagrantfile
     login           
     package         packages a running vagrant environment into a box
     plugin          manages plugins: install, uninstall, update, etc.
     port            displays information about guest port mappings
     powershell      connects to machine via powershell remoting
     provision       provisions the vagrant machine
     push            deploys code in this environment to a configured destination
     rdp             connects to machine via RDP
     reload          restarts vagrant machine, loads new Vagrantfile configuration
     resume          resume a suspended vagrant machine
     snapshot        manages snapshots: saving, restoring, etc.
     ssh             connects to machine via SSH
     ssh-config      outputs OpenSSH valid configuration to connect to the machine
     status          outputs status of the vagrant machine
     suspend         suspends the machine
     up              starts and provisions the vagrant environment
     upload          upload to machine via communicator
     validate        validates the Vagrantfile
     version         prints current and latest Vagrant version
     winrm           executes commands on a machine via WinRM
     winrm-config    outputs WinRM configuration to connect to the machine

For help on any individual command run `vagrant COMMAND -h`

Additional subcommands are available, but are either more advanced
or not commonly used. To see all subcommands, run the command
`vagrant list-commands`.

vagrant@vagrant:~$
```

If you are having issues, ensure that any vagrant package is fully uninstalled and then install the official package again. It may be that a `vagrant` executable is in your PATH that should not be. After uninstalling vagrant, it should be easy to fine with a simple `which vagrant` or `whereis vagrant`.

Cheers!

**Comment by briancain** [comment]
@ginsburgnm - Installing Vagrant 2.2.0 works for me on Ubuntu 18.04.1 LTS:

```
vagrant@vagrant:~$ cat /etc/lsb-release
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=18.04
DISTRIB_CODENAME=bionic
DISTRIB_DESCRIPTION="Ubuntu 18.04.1 LTS"
vagrant@vagrant:~$ curl -O https://releases.hashicorp.com/vagrant/2.2.0/vagrant_2.2.0_x86_64.deb
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 38.2M  100 38.2M    0     0  5680k      0  0:00:06  0:00:06 --:--:-- 5338k
vagrant@vagrant:~$ sudo dpkg -i vagrant_2.2.0_x86_64.deb
Selecting previously unselected package vagrant.
(Reading database ... 42019 files and directories currently installed.)
Preparing to unpack vagrant_2.2.0_x86_64.deb ...
Unpacking vagrant (1:2.2.0) ...
Setting up vagrant (1:2.2.0) ...
vagrant@vagrant:~$ vagrant --help
WARNING: This command has been deprecated in favor of `vagrant cloud auth login`
Usage: vagrant [options] <command> [<args>]

    -v, --version                    Print the version and exit.
    -h, --help                       Print this help.

Common commands:
     box             manages boxes: installation, removal, etc.
     cloud           manages everything related to Vagrant Cloud
     destroy         stops and deletes all traces of the vagrant machine
     global-status   outputs status Vagrant environments for this user
     halt            stops the vagrant machine
     help            shows the help for a subcommand
     init            initializes a new Vagrant environment by creating a Vagrantfile
     login
     package         packages a running vagrant environment into a box
     plugin          manages plugins: install, uninstall, update, etc.
     port            displays information about guest port mappings
     powershell      connects to machine via powershell remoting
     provision       provisions the vagrant machine
     push            deploys code in this environment to a configured destination
     rdp             connects to machine via RDP
     reload          restarts vagrant machine, loads new Vagrantfile configuration
     resume          resume a suspended vagrant machine
     snapshot        manages snapshots: saving, restoring, etc.
     ssh             connects to machine via SSH
     ssh-config      outputs OpenSSH valid configuration to connect to the machine
     status          outputs status of the vagrant machine
     suspend         suspends the machine
     up              starts and provisions the vagrant environment
     upload          upload to machine via communicator
     validate        validates the Vagrantfile
     version         prints current and latest Vagrant version
     winrm           executes commands on a machine via WinRM
     winrm-config    outputs WinRM configuration to connect to the machine

For help on any individual command run `vagrant COMMAND -h`

Additional subcommands are available, but are either more advanced
or not commonly used. To see all subcommands, run the command
`vagrant list-commands`.

vagrant@vagrant:~$ vagrant --version
Vagrant 2.2.0
```

**Comment by ginsburgnm** [comment]
I'll just assume there's a conflict with something my org is doing. because, same version of vagrant and ubuntu. different results. 
$: apt list --installed | grep vagrant
WARNING: apt does not have a stable CLI interface. Use with caution in scripts.
vagrant/now 1:2.2.0 amd64 [installed,local]

**Comment by chrisroberts** [comment]
@ginsburgnm The output you provided above shows this path in the error message:

```
/usr/local/bin/vagrant
```

which then goes on to mention the path:

```
/usr/lib/ruby/2.5.0/rubygems.rb
```

The Vagrant executable at `/usr/local/bin/vagrant` is not something that our Vagrant package provides and is likely a remnant of the system vagrant package that you installed which was not cleaned up correctly. Removing that file should get the correct Vagrant executable to be picked up.

**Comment by vranki** [comment]
I created a bug in Ubuntu bug tracker: https://bugs.launchpad.net/ubuntu/+source/vagrant/+bug/1804626

**Comment by maratbn** [comment]
In my case this was caused by trying to install Vagrant from source on an Ubuntu 18.04 system, and then abandoning that attempt and trying to go back to the regular Ubuntu Vagrant.

I think that installed a bunch of extraneous Ruby gems that caused conflicts with the regular Ubuntu Vagrant.

The way I found these incompatible gems is by running this command to find gems not tracked by the package manager:

`comm -23 <(find / -xdev -type f|sort) <(sort /var/lib/dpkg/info/*.list) | grep gem`

This told me I had a whole bunch of untracked gems under `/var/lib/gems/2.5.0`.  So I got rid of the `2.5.0` directory, and the regular Ubuntu Vagrant started working fine.
